iquip/apps/logger.py

```diff
@@ -125,52 +136,71 @@ class LoggerApp(qiwis.BaseApp):
+    def __init__(self, name: str = "logger", path: str = "logs", parent: Optional[QObject] = None):
         """Extended."""
         super().__init__(name, parent=parent)
         self.loggerFrame = LoggerFrame()
         self.confirmFrame = ConfirmClearingFrame()
         # connect signals to slots
         self.loggerFrame.clearButton.clicked.connect(self.checkToClear)
         self.confirmFrame.confirmed.connect(self.clearLog)
-        self.handler = LoggingHandler(self.addLog)
-        # TODO(aijuh): Change the log format when it is determined.
-        fs = "%(levelname)s [%(name)s] [%(filename)s:%(lineno)d] %(message)s"
-        formatter = logging.Formatter(fs)
-        self.handler.setFormatter(formatter)
+        # initialize handlers
+        self.levelsDict = {logging.DEBUG: "DEBUG", logging.INFO: "INFO", logging.WARNING: "WARNING",
+                           logging.ERROR: "ERROR", logging.CRITICAL: "CRITICAL"}
+        self.frameHandler = LoggingHandler(self.addLog)
+        logFileName = path + QDateTime.currentDateTime().toString("yyMMdd-HHmmss")
+        self.fileHandler = handlers.TimedRotatingFileHandler(filename=logFileName, when="midnight",
+                                                             interval=1, encoding="utf-8")
+        self.initLogger()
+        # set loggerFrame's frameLevelBox
+        frameLevelBox = self.loggerFrame.frameLevelBox
+        frameLevelBox.addItems(self.levelsDict.values())
+        frameLevelBox.textActivated.connect(partial(self.setLevel, self.frameHandler))
+        frameLevelBox.setCurrentText(self.levelsDict[self.frameHandler.level])
+        # set loggerFrame's fileLevelBox
+        fileLevelBox = self.loggerFrame.fileLevelBox
+        fileLevelBox.addItems(self.levelsDict.values())
+        fileLevelBox.textActivated.connect(partial(self.setLevel, self.fileHandler))
+        fileLevelBox.setCurrentText(self.levelsDict[self.fileHandler.level])
+
+    def initLogger(self):
+        """Initializes the root logger and handlers for constructor."""
+        self.fileHandler.suffix = '-(%Y%m%d)'
+        shortFormat = "[%(name)s] %(message)s"
+        longFormat = "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] %(message)s"
+        self.frameHandler.setFormatter(logging.Formatter(shortFormat))
+        self.fileHandler.setFormatter(logging.Formatter(longFormat))
+        # set rootLogger
         rootLogger = logging.getLogger()
-        rootLogger.addHandler(self.handler)
-        self.setLevel("WARNING")
-        self.loggerFrame.levelBox.addItems(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"])
-        self.loggerFrame.levelBox.textActivated.connect(self.setLevel)
-        self.loggerFrame.levelBox.setCurrentText("WARNING")
+        rootLogger.addHandler(self.frameHandler)
+        rootLogger.addHandler(self.fileHandler)
+        self.frameHandler.setLevel("WARNING")
+        self.fileHandler.setLevel("WARNING")
+        rootLogger.setLevel("WARNING")
```

Comment:
I am sorry, you don't have to set its level since `setLevel()` sets the root logger's level as well.